tests/unit/test_search_model.py

```diff
@@ -1558,3 +1569,29 @@ def test_with_active_term():
     assert isinstance(term4, Term) is True
     assert term4.field == "__glossary"
     assert term4.value == GLOSSARY_QUALIFIED_NAME
+
+
+def test_dsl_serialization_and_deserialization():
+
+    dsl1 = DSL(
+        from_=0,
+        aggregations={"test1": {"test2": {"field": "__test_field"}}},
+        size=300,
+        sort=[SortItem(field="created", order=SortOrder.ASCENDING, nested_path="test")],
+        query=Term(field="__typeName.keyword", value="Schema"),
+        track_total_hits=False,
+    )
+
+    data = {
+        "aggregations": {"test1": {"test2": {"field": "__test_field"}}},
+        "query": {"term": {"__typeName.keyword": {"value": "Schema"}}},
+        "from": 0,
+        "size": 300,
+        "track_total_hits": False,
+        "sort": [{"created": {"order": "asc", "nested": {"path": "test"}}}],
+    }
+    test_dsl = DSL(**data)
```

Comment:
Let's also ensure we test this on complex data structures in unit tests, e.g., nested `aggregations`, multiple `sort` items in a list, and multiple queries of different types (`term`, `bool`, `range`, etc).

```suggestion
   raw_dsl_data = {
        "aggregations": {"test1": {"test2": {"field": "__test_field"}}},
        "query": {"term": {"__typeName.keyword": {"value": "Schema"}}},
        "from": 0,
        "size": 300,
        "track_total_hits": False,
        "sort": [{"created": {"order": "asc", "nested": {"path": "test"}}}],
    }
   dsl_through_raw = DSL(**raw_dsl_data)
```

Reply:
Done